### PR TITLE
Day 11

### PR DESCRIPTION
**Pull request by Yag000**

### CODE REVIEW

File: contents/reductions/cayley-hamilton.tex [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+
+\begin{proof}
+	Soit $x \in E$ non nul. $E_{u,x} = \vect(u^k(x))$ stable par $u$.
+
+	%TODO
+\end{proof}
+
+
+\subsubsection{Sous espaces caractéristiques}
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
\subsubsection{Sous-espaces caractéristiques}
```

File: contents/reductions/cayley-hamilton.tex [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+\begin{coro}
+	Soit $u \in \LE$ tel que $\chi_u$ est scindé, i.e. $\chi_u = \prod_{i=1}^r (X - \lambda_i)^{m_i}$.
+	Alors $E$ est la somme directe des sous espaces caractéristiques de $u$. En particulier, a dimension du
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
	Alors $E$ est la somme directe des sous-espaces caractéristiques de $u$. En particulier, la dimension du
```

File: contents/reductions/cayley-hamilton.tex [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	sous espace caractéristique est égale a la multiplicité de la valeur propre associée.
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
	sous-espace caractéristique est égale à la multiplicité algébrique de la valeur propre associée.
```

File: contents/reductions/cayley-hamilton.tex [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+\end{coro}
+
+\begin{proof}
+	$\chi_u(X) = \prod_{i=1}^r (X - \lambda_i)^{m_i}$ où les $\lambda_i$ sont distincts, avec $P_i(X) = (X - \lambda_i)^{m_i}$.
+	Les $P_i$ sont premiers entre eux. D'après le lemme des noyaux, on a que $E = \bigoplus_{i=1}^r \ker P_i(u)$.
+
+	Soit $v_j = \restr u {\ker P_j(u)}$. On a que $\chi_u = \prod_{j=1}^r \chi_{v_j}$. Or la seule racine de $P_i$ est $\lambda_i$,
+	$u$ annule $P_i$ et $\deg P_i = m_i$. Donc $m_i$ est bien la dimension du sous espace caractéristique associé à $\lambda_i$.
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
	$u$ annule $P_i$ et $\deg P_i = m_i$. Donc $m_i$ est bien la dimension du sous-espace caractéristique associé à $\lambda_i$.
```

File: contents/reductions/cayley-hamilton.tex [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+$\mu_u \in  \set {(X-1), (X-1)^2, (X-1)^3}$
+
+$M - I_3 = \begin{pmatrix}
+		0 & 1 & 0 \\ 0 & 0 & 0 \\ 0 & 0 & 0
+	\end{pmatrix} \neq 0$ Donc $\mu_u \neq X - 1$.
+$(M - I_3)^2 = 0$ donc $\mu_u = (X-1)^2$, d'où $m_m(1) = 2$.
+
+
+$m_g(\lambda)$ la multiplicité géométrique de $\lambda$ et est la dimension du sous espace caractéristique associé à $\lambda$.
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
$m_g(\lambda)$ la multiplicité géométrique de $\lambda$ est la dimension du sous-espace caractéristique associé à $\lambda$.
```

File: contents/reductions/cayley-hamilton.tex [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+$E_1 (M) = \kre (u-id) = \vect (e_1, e_3)$ donc $m_g(1) = 2$.
+
+
+\begin{prop}
+	Soit $u \in \LE$ et $\lambda$ une valeur propre de $u$. On sait que $ 1 \geq m_g(\lambda) \geq m_a(\lambda)$ et $1 \geq m_m(\lambda) \geq m_a(\lambda)$.
+
+	Ceci est une consequence du lemme suivant.
```
Comment (ErinLeb) [inline contents/reductions/cayley-hamilton.tex]:
```suggestion
	Ceci est une conséquence du lemme suivant.
```

File: contents/reductions/compagnie.tex [hunk 7]
```diff
@@ -0,0 +1,94 @@
+\subsection{Matrices compagnies}
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
\subsection{Matrices compagnons}
```

File: contents/reductions/compagnie.tex [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+
+
+\begin{definition}
+	La matrice compagnie ou matrice de Frobenius d'un polynôme unitaire $P(X)  = X^n - \sum_{k=0}^{n-1} a_k X^k$ est la matrice
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	La matrice compagnon ou matrice de Frobenius d'un polynôme unitaire $P(X)  = X^n - \sum_{k=0}^{n-1} a_k X^k$ est la matrice
```

File: contents/reductions/compagnie.tex [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+			0      & 0      & \ldots & 0      & a_0     \\
+			1      & 0      & \ldots & 0      & a_1     \\
+			0      & 1      & \ldots & 0      & a_2     \\
+			\vdots & \vdots & \ddots & \vdots & \vdots  \\
+			0      & 0      & \ldots & 1      & a_{n-1}
+		\end{pmatrix}$$
+\end{definition}
+
+\begin{remarque}
+	Soit $u$ l'endomorphise de $E$ tel que $C_p$ soit la matrice de $u$ dans une base $\mathcal{B} = (e_1, \ldots, e_n)$.
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	Soit $u$ l'endomorphisme de $E$ tel que $C_p$ soit la matrice de $u$ dans une base $\mathcal{B} = (e_1, \ldots, e_n)$.
```

File: contents/reductions/compagnie.tex [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+\end{remarque}
+
+\begin{proof}
+	%TODO finish proof
+	\begin{itemize}
+		\item Soit $x \in E$ tel que $E_{u,x} = E$, et on choisit $\mathcal{B} = (x, u(x), \dots, u^{n-1}(x))$.
+	\end{itemize}
+\end{proof}
+
+\subsubsection{Matrice caractéristique }
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
\subsubsection{Polynôme caractéristique }
```

File: contents/reductions/compagnie.tex [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+\begin{prop}
+	Le polynôme caractéristique de $C_P$, la matrice compagnon associée à $P$ est $P$.
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	Le polynôme caractéristique de $C_P$, la matrice compagnon associée à $P$, est $P$.
```

File: contents/reductions/compagnie.tex [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+	$\Delta(a_1, \ldots, a_n)(X) = \det (XI_n - C_P) =
+		\begin{vmatrix}
+			X      & 0      & \ldots & 0      & -a_0      \\
+			-1     & X      & \ldots & 0      & -a_1      \\
+			0      & -1     & \ldots & 0      & -a_2      \\
+			\vdots & \vdots & \ddots & \vdots & \vdots    \\
+			0      & 0      & \ldots & -1     & X-a_{n-1}
+		\end{vmatrix}$
+
+	On développe par rapport a la premiere ligne.
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	On développe par rapport à la premiere ligne.
```

File: contents/reductions/compagnie.tex [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+	%TODO: Explanation ??
+	$$ \Delta(a_1, \ldots, a_n)(X) = (-1)^{1+1}X\Delta(a_1, \ldots, a_n-1)(X) + (-1)^{1+n}(-1)^{n-1}a_0$$
+	et donc $\Delta(a_1, \ldots, a_n)(X) = X\Delta(a_1, \ldots, a_n-1)(X) - a_0 $
+	Et par itération on obtiens $\Delta(a_1, \ldots, a_n)(X) = X^n - \sum_{k=0}^{n-1} a_k X^k$
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	Et par itération on obtient $\Delta(a_1, \ldots, a_n)(X) = X^n - \sum_{k=0}^{n-1} a_k X^k$
```

File: contents/reductions/compagnie.tex [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+
+\begin{prop}
+	Soit $u \in \LE$ cyclique et $x \in E$ tel que $E_{u,x} = E$. On choisit les $a_k \in $ tels que
+	$$u ^(x) = \sum_{k=0}^{n-1} a_k u^k(x)$$
+	Alors $\chi_u (X) = X^n - \sum_{k=0}^{n-1} a_k X^k$
+\end{prop}
+
+\begin{remarque}
+	Un corollaire direct de ce énoncé dit que tout polynôme unitaire de degré $n$ dans $\K[X]$ est le polynôme caractéristique
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	Un corollaire direct de cet énoncé dit que tout polynôme unitaire de degré $n$ dans $\K[X]$ est le polynôme caractéristique
```

File: contents/reductions/compagnie.tex [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+\begin{proof}
+	Cela découle du fait que dans $(x, u(x), \ldots, u^{n-1}(x))$ la matrice de $u$ est une matrice compagnon associée à un $P \in \K[X]$.
+	$P$ est construit pour que $P(u)(x) = 0_E$.\\
+	On a $P(u) = 0_{\LE}$ et $\chi_u = P$.
+\end{proof}
+
+
+\begin{example}
+	Une condition nécessaire et suffisante pour qu'un endomorphisme nilpotent soit cyclique est
+	que son indice de nilpotence soit égal à a $n = \dim E$.
```
Comment (ErinLeb) [inline contents/reductions/compagnie.tex]:
```suggestion
	que son indice de nilpotence soit égal à $n = \dim E$.
```

**Review by ErinLeb — COMMENTED** [review]
You should rename `compagnie.tex`

**Comment by Yag000** [comment]
> You should rename `compagnie.tex`

Done, thx :)

**Pull request merged**